lib/modules/dosomething/dosomething_global/dosomething_global.module

```diff
@@ -413,146 +461,76 @@ function dosomething_global_get_language($account, stdClass $node = NULL) {
-      }
-    }
+    // Build complete URL and redirect
+    $new_url = str_replace(' ', '', '/node/' . $node->nid . '/edit/' . $lang_append);
+    drupal_goto($new_url);
   }
-  return FALSE;
 }
 
 /**
- * Handle redirects to translated content for authenticated users. In this
- * case:
+ * Handle redirects to translated content
  *
- * - If my profile includes a language preference,
- * - and I'm not a privileged user (admin, editor, etc.),
- * - and I request a bare URL (e.g., /campaigns/sample),
- * - and there is a translation available (e.g., /voluntario/ejemplo),
- * - then issue a 302 redirect to the translation's URL.
- */
-function dosomething_global_redirect_node_auth() {
-  global $user;
-
-  // Don't do redirects for privileged users.
-  if (user_access('access administration menu')) {
-    return;
-  }
-
-  // User language.
-  $user_lang_code = $user->language ?: 'en';
-
-  // All languages, as an array of objects.
-  $languages = language_list();
-
-  $global_lang_code = DOSOMETHING_GLOBAL_DEFAULT_LANG_CODE;
-
-  if (empty($languages[$global_lang_code])) {
-    watchdog('dosomething_global', "Can't load %lang language",
-      array('%lang' => $global_lang_code), WATCHDOG_ERROR);
-    return;
-  }
-
-  // Don't redirect for Global English speakers.
-  if ($user_lang_code == $global_lang_code) {
-    return;
-  }
-
-  $node = node_load(arg(1));
-
-  if ($languages[$user_lang_code]) {
-    dosomething_global_redirect_node_language($node, $user_lang_code, 302);
-  }
-}
-
-/**
- * Handle redirects to translated content for anonymous users. In this
- * case:
+ * @param $user_lang
+ *   The users language
+ * @param $node
+ *   The specified node to redirect too
+ * @param $redirect_status
+ *   The status to redirect with. 301 for anonymous, 302 for authenticated
  *
- * - If I request a bare URL (e.g., /campaigns/sample),
- * - and there is a translation available (e.g., /voluntario/ejemplo),
- * - then issue a 301 redirect to the translation's URL.
+ * If the user is an administrator or global english speaker
+ *  - Do not redirect
+ * If the node has a translation available for the specified language
+ *  - Redirect them to the translation
+ * Else
+ *  - 404 the user
  */
-function dosomething_global_redirect_node_anon() {
+function dosomething_global_redirect_node($user_lang, $node, $redirect_status) {
 
-  // All languages, as an array of objects.
-  $languages = language_list();
-
-  // Map GeoIP country to a system language. Default to 'en-global'.
-  $user_country_code = dosomething_settings_get_geo_country_code();
-  $target_lang_code = DOSOMETHING_GLOBAL_DEFAULT_LANG_CODE;
-
-  $lang_map = variable_get('dosomething_global_language_map', '');
-  foreach ($lang_map as $lang_key => $lang) {
-    if ($lang['country'] == $user_country_code) {
-      $target_lang_code = $lang_key;
-      break;
+  // Don't do redirects for privileged users.
+  if ($redirect_status == 302) {
+    if (user_access('access administration menu')) {
```

Comment:
I don't see this in the previous version, this means anyone on our staff (admin, editor, site admin, etc) and regional admins won't get redirected here.. is that intended?